tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala

```diff
@@ -1,37 +1,47 @@
 package at.forsyte.apalache.tla.lir.transformations.standard
 
-import at.forsyte.apalache.tla.lir.{LetInEx, OperEx, TlaEx}
-import at.forsyte.apalache.tla.lir.transformations.{LanguageWatchdog, TlaExTransformation, TransformationTracker}
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.transformations.{TlaExTransformation, TransformationTracker}
 
-class DeepCopy(tracker: TransformationTracker) extends TlaExTransformation {
-  override def apply(expr: TlaEx): TlaEx = {
-    transform(expr)
+/**
+  * DeepCopy constructs a structurally identical copy of a given TlaEx or TlaDecl, with fresh unique IDs.
+  */
+class DeepCopy( tracker : TransformationTracker ) {
```

Comment:
why do you introduce `deepCopyEx` instead of implementing `TlaExTransformation`? You could implement both `TlaExTransformation` and `TlaDeclTransformation`. (I am going to merge #444 as soon as it passes all the tests).

Reply:
I was debating this internally. The reason I went in this direction is exactly that: `DeepCopy` would extend two transformations and therefore have two overloaded `apply` methods, which you couldn't visually tell apart in use cases. This implementation, I think, makes it impossible to confuse the two methods.

Reply:
I see. The `apply` methods may be confusing.